### PR TITLE
core: Various small Font and Glyph refactors

### PR DESCRIPTION
**Pull request by Dinnerbone**
Dependencies for TTF rendering

### CODE REVIEW

File: core/src/font.rs [hunk 1]
```diff
@@ -60,36 +60,94 @@ impl EvalParameters {
...
     scale: f32,
 
-    /// Kerning infomration.
-    /// Maps from a pair of unicode code points to horizontal offset value.
-    kerning_pairs: fnv::FnvHashMap<(u16, u16), Twips>,
-
     /// The distance from the top of each glyph to the baseline of the font, in
     /// EM-square coordinates.
-    ascent: u16,
+    ascent: i16,
```
Comment (sleepycatcoding) [inline core/src/font.rs:146]: Out of interest, why is this change necessary, can the ascent be negative?

File: core/src/font.rs [hunk 2]
```diff
@@ -418,38 +469,78 @@ impl<'gc> Font<'gc> {
...
+        }
+    }
+}
+
 #[derive(Debug, Clone)]
 pub struct Glyph {
     // Handle to registered shape.
     // If None, it'll be loaded lazily on first render of this glyph.
-    shape_handle: RefCell<Option<ShapeHandle>>,
+    // It's a double option; the outer one is "have we registered", the inner one is option because it may not exist
```
Comment (sleepycatcoding) [inline core/src/font.rs:522]: I think this should be doc comment.

**Review by sleepycatcoding — APPROVED** [review]
I don't have experience in handling fonts, but this looks good to me.

File: core/src/font.rs (hunk 1 above)
Comment (Dinnerbone) [reply core/src/font.rs:146]: For TTFs yes, apparently

File: core/src/font.rs [hunk 3]
```diff
@@ -519,14 +519,15 @@ impl GlyphShape {
...
+    shape_handle: RefCell<Option<Option<ShapeHandle>>>,
```
Comment (adrian17) [inline core/src/font.rs:523]: to be sure, there currently doesn't exist a case where we put a `None` here - it's for a future PR?

**Review by adrian17 — APPROVED** [review]

File: core/src/font.rs (hunk 3 above)
Comment (Dinnerbone) [reply core/src/font.rs:523]: Yeah, font files can define properties of a glyph without any way to display it - they'll be whitespace characters

Comment (torokati44) [reply core/src/font.rs:523]: Wouldn't this be nicer as an `enum`?
